### PR TITLE
Change get_logits to return a single slice

### PR DESCRIPTION
**Pull request by jcsoo**
Per our discussion in #49, I originally implemented `get_logits(row: usize) -> Result<Option<&[f32]>, WhisperError>` but struggled with the documentation. 

The problem is that we don't know anything about how the logits matrix might be structured in the future, so the only thing we can say is that the row parameter must be zero. Because of that, I can't think of a current use case where the caller could usefully pass in a row number that was non-zero.

We also don't know whether the parameter zero would actually be the right value to use in the future. For instance, if whisper.cpp changes to returning the whole logits matrix, getting the logits for the last token might require `get_logits(num_tokens-1)` or redefining that the rows are indexed in reverse order.

So instead I ended up implemented the most basic version of `get_logits()` possible that returns a &[f32] slice of length n_vocab, documenting our current understanding of what it provides. If a future version of whisper.cpp allows retrieving a full 2D logits matrix, additional functions such as `get_logits_matrix()` and `get_logits_row(row: usize)` can be added.

### CODE REVIEW

**Comment by tazz4843** [comment]
Will review when I have access to a computer later tonight.

**Review by tazz4843 — APPROVED** [review]
